docs/docs/api/index.mdx

```diff
@@ -5,3 +5,35 @@ title: API reference
 These docs aim to cover the entire public surface of the core dagster APIs, as well as public APIs from all provided libraries.
 
 Dagster follows [semantic versioning](https://semver.org/). We attempt to isolate breaking changes to the public APIs to minor versions on a roughly 12-week cadence, and will announce deprecations in Slack and in the release notes to patch versions on a roughly weekly cadence.
+
+## AssetSpec
```

Comment:
I'll let others weigh in on the code changes, but this page shouldn't have these kinds of docs -- in this case, I'd recommend adding content to this page: https://docs.dagster.io/guides/build/assets/metadata-and-tags/kind-tags